presto-main/src/main/java/com/facebook/presto/server/QueryStateInfoResource.java

```diff
@@ -66,19 +70,30 @@ public QueryStateInfoResource(
 
         return queryInfos.stream()
                 .filter(queryInfo -> !queryInfo.getState().isDone())
-                .map(queryInfo -> createQueryStateInfo(queryInfo, queryManager.getQueryResourceGroup(queryInfo.getQueryId()).map(resourceGroupManager::getResourceGroupInfo)))
+                .map(this::getQueryStateInfo)
                 .collect(toImmutableList());
     }
 
+    private QueryStateInfo getQueryStateInfo(QueryInfo queryInfo)
+    {
+        Optional<ResourceGroupId> groupId = queryManager.getQueryResourceGroup(queryInfo.getQueryId());
+        if (queryInfo.getState() == QUEUED) {
```

Comment:
So we roll back to the previous behavior that only QUEUED queries has resource group info?

Reply:
Yes.